File: 2026-08/src/day_25/prms_120840/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(int balls, int share) {
    long long ans = 1;
    for (int i = 1; i <= share; i++) {
        ans = ans * (balls - i + 1) / i;
    }

    return (int)ans;
}

File: 2026-08/src/day_25/prms_120840/Solution.java
package day_25.prms_120840;

class Solution {
    public int solution(int balls, int share) {
        long ans = 1;
        for (int i = 1; i <= share; i++) {
            ans = ans * (balls - i + 1) / i;
        }

        return (int) ans;
    }
}
